[REF] website_variants_extra: place product price, product qty, and "add to cart "btn in  3rd column. It also add the reviews number, Icons(quote, Ask question,Print) and the behavior to share products . The product name change acording to the selected variants.  HU#1006 ISSUE#571

gmlp opened the pull request:
**BUILD:**  https://github.com/Vauxoo/yoytec/pull/675
@oscarolar, these are the changes applied in website_variants_extra module.
Best regards

oscarolar commented:
Reference this PR to the plan please.
and create a dummy build with yoytec env.

nhomar commented:
@gmlp 
1. Your commit meessage says "website_yoytec:" which is a module that not belong to this PR.

Commit:
[REF] website_variants_extra: layout add a 3rd column where the
price,product qty and "add to cart btn" are moved from 2nd to 3rd column.
it also add product availability in 3rd column.

gmlp commented:
![screenshot from 2016-02-15 13-08-37](https://cloud.githubusercontent.com/assets/10885517/13057975/61668044-d3e5-11e5-87d5-46859ba74f04.png)
@oscarolar

oscarolar commented:
![asfd](https://www.evernote.com/l/AFyh8ABVDIxBK7GgyM7klgNwG9DAtr6ykEEB/image.png)

oscarolar commented:
I cant see the manufacturer model and the SKU, did you miss it?

oscarolar commented:
What was wrong with the icons the customer provided?
![sad](https://www.evernote.com/l/AFxmPAFQJyZAR70xJ0DwyuT1iejF7ZHgjMwB/image.png)

oscarolar commented on website_variants_extra/views/templates.xml:
```diff
@@ -45,12 +46,125 @@
+                                    </a>
+                                </span>
+                            </div>
+                            <a id="add_to_cart" class="btn btn-warning btn-lg mt16 js_check_product a-submit" href="#">Add to Cart</a>
```
Can you remove the class `btn-lg` to make the button fit the size asked by the customer.

oscarolar replied:
Not valid anymore.

gmlp commented:
@oscarolar , the icons asked don't appears in https://fortawesome.github.io/Font-Awesome/icons/

oscarolar commented:
A lot of space between text on availability, set the `line-height` style to a smaller value.
![s](https://www.evernote.com/l/AFzUMh_ln25DuL6B4TXM2fhDIO7dH47k8rEB/image.png)

oscarolar commented:
I still see a pill on the 'Extra price' section, as the customer asked it must be a rectangle
![nopill](https://www.evernote.com/l/AFz-gudGw59GKqgAdPLA3aOs9KdhTqTN2SwB/image.png)

oscarolar commented:
I see black some text that should be gray
![nopill](https://www.evernote.com/l/AFyBoofgG9JC3JGbupWoka_l0ElLYNV061EB/image.png)

oscarolar commented on website_variants_extra/views/templates.xml:
```diff
@@ -45,12 +46,125 @@
+                                        <td><span class="sale-action">Add To Wishlist</span></td>
+                                    </tr>
+                                    <tr>
+                                        <td class="text-center"><span class="fa fa-question"/></td>
```
use `fa fa-question-circle`

oscarolar commented:
I cant see the share icons, I have mentioned before that there is a simple implementation on how to sahre the  product, since there are only two lines of code to add in order to get this working I suggest that you only copy the lines of [this module](https://github.com/OCA/e-commerce/blob/8.0/website_product_share/views/website_product.xml#L8-L9) instead of adding a dependency and installing an extra module.
![shareicons](https://www.evernote.com/l/AFw6y7TdcGxFbr_r1bjPqj8BNWKtelisbM4B/image.png)

oscarolar commented:
I cant see the breadcrumb on your PR, what happened?
![no](https://www.evernote.com/l/AFyYKSKKXC1PN7cnkT-3xo8OJDp01qqbh9wB/image.png)

oscarolar commented on website_rate_product/__openerp__.py:
```diff
@@ -7,7 +7,7 @@
     "website": "http://www.vauxoo.com/",
     "license": "AGPL-3",
     "depends": [
-        "website_sale"
```
why did you remove this and added the other one?

gmlp commented:
@oscarolar sku and manufacturer belong to another repo

oscarolar commented on website_variants_extra/static/src/less/website.less:
```diff
@@ -0,0 +1,32 @@
+.js_quantity.form-control {
+    background-color: white;
+
+}
```
please put all these classes namespaced, do not leave them just as simple CSS rules, this mean you have to nest them under a parent selector.

oscarolar commented on website_variants_extra/views/templates.xml:
```diff
@@ -45,12 +46,125 @@
               </t>
-		  </xpath> TODO: THIS CODE IS COMMENTED BECAUSE IT CRASHES PHANTOM CRAWLER, 
-		  				 IT HAS A PURPOSE DISCUSS WITH NHOMAR--> 
+		  </xpath> TODO: THIS CODE IS COMMENTED BECAUSE IT CRASHES PHANTOM CRAWLER,
```
This is no longer happening, please delete the code

gmlp commented on website_rate_product/views/star_rate.xml:
```diff
@@ -9,8 +9,8 @@
 
-        <template id="rating_product" inherit_id="website_sale.product" active="False" customize_show="True" name="Show Product Rating">
-            <xpath expr="//h1[@itemprop='name']" position="after">
+        <template id="rating_product" inherit_id="website_variants_extra.product_variants_description_3columns" active="False" customize_show="True" name="Show Product Rating">
```
why did you remove this and added the other one?
@oscarolar ,Because, i replace all the right div and change the target to website_variants_extra, therefore this module depends of website_variants_extra

oscarolar replied:
I'm talking about dependencies you removed the dependency `website_sale`and added `website_variants_extra`, this has nothing to do with the divs or whatever, besides, if I want to install the module `website_rate_product`only, I can't, you have to inherit `website_rate_product` from `website_variants_extra` and then do the correct inheritance. https://github.com/Vauxoo/addons-vauxoo/pull/708/files#diff-2aa691e2f5886e33884e188617cd0711L10

gmlp commented:
![screenshot from 2016-02-16 17-07-30](https://cloud.githubusercontent.com/assets/10885517/13094377/e40e8cd6-d4cf-11e5-8cd9-08bbbe36be90.png)

@oscarolar , dude change done

nhomar commented:
@gmlp Please answer the first @oscarolar request:

Create a Dommy PR on customer repository and reference it here, Please.

gmlp commented:
@nhomar,
**workplan:** https://github.com/vauxoo/yoytec/pull/640
**dummy:**  https://github.com/Vauxoo/yoytec/pull/675

oscarolar commented:
@gmlp regarding the icon theme please use the ones provided by the customer https://github.com/Vauxoo/yoytec/issues/571

gmlp commented:
@oscarolar , changes done.
![screenshot from 2016-02-18 11-47-58](https://cloud.githubusercontent.com/assets/10885517/13152595/8ee89b0e-d635-11e5-8941-a986df3171ec.png)

gmlp commented:
![screenshot from 2016-02-18 18-34-06](https://cloud.githubusercontent.com/assets/10885517/13162952/5e7618b4-d66e-11e5-8c1d-1a5733e135a2.png)
@oscarolar , improvements done.

oscarolar commented on website_variants_extra/static/src/js/website_sale.js:
```diff
@@ -1,48 +1,38 @@
+            if(index > -1){
+                $product_name = $product_name.substring(0,index);
+            }
+            $('#product_with_variants').html($product_name+", "+variants_str);
```
this JS must be tested with PhantomJS, please make a test that change selected attributes and verfy that it build the name correctly.

oscarolar commented on website_variants_extra/static/src/js/website_sale.js:
```diff
@@ -1,48 +1,38 @@
+        var comments_qty = $('#comments-list').children().length;
+        if (comments_qty > 0){
+            var concept = comments_qty >= 2 ? "reviews" : "review";
+            $('#no-reviews').text(comments_qty+" "+concept);
```
what if there are comments in the backend? this algorithm will work?, take in account that if you post a comment in the backend it will be rendered on the frontend even   (unpublished), I don't see any validation for unpublished comments.

gmlp replied:
@oscarolar, it's only happens with  super users accounts that select which comments be posted.

Commit:
[ADD] website_variants_extra: PhantomJS tests and demo data added.

oscarolar commented on website_product_availability/models/product.py:
```diff
@@ -151,11 +151,12 @@ def _get_planned_dates(self):
         domain_quant = []
         product = self
         domain_quant += [('product_id', 'in', [product.id])]
+        # pylint: disable=unused-variable
```
This was made because the method returns 3 variables but we need only one.

nhomar replied:
Si no está siendo usada borrala

oscarolar commented:
:+1:

nhomar commented on website_product_availability/models/product.py:
```diff
@@ -151,11 +151,12 @@ def _get_planned_dates(self):
     def _product_availability_warehouse(self, warehouse_id):
         # search avalailability for stokable products
-        if self.type not in('product'):
+        if self.type not in 'product':
```
el in es para comparar iterables, haber usado una tupla no está mal.... o usas == o usas in con un iterable. ambas opciones son válidas, tu cambio está incorrecto.

nhomar commented on website_rate_product/models/mail_message.py:
```diff
@@ -27,7 +27,7 @@
 class MailMessage(osv.Model):
     _inherit = 'mail.message'
-    _columns = {
+    _columns = {  # pylint: disable=attribute-deprecated
```
usa nueva api.

los pylint disabled es cuando no se puede resolver algo, evitenlos a lo máximo por favor, esto es estúpido que sea hecho de eta manera.

gmlp commented:
@oscarolar and @nhomar ,changes done

nhomar commented:
@gmlp Can you please before merge:

1.- Rebase this with base addons-vauxo.
2.- What I understand in the first 2 commit one of them fixe the other one. (or at least you mention something which I did not understand was "Impossible".
3.- Please update de description of the PR in order to si the Before - After with this feature, to many messages in the middle make to me impossible to understand the real conclusion, make a good resume of the job achieved with this PR.
4.- You mentiondes an HU in the title of the PR but I can not see a mention the #CA which you are covering.

Thanks in advance.

gmlp commented:
@oscarolar and @nhomar ,
Dummy: https://github.com/Vauxoo/yoytec/pull/675
Berofe:
![screenshot from 2016-03-01 11-53-11](https://cloud.githubusercontent.com/assets/10885517/13436332/548d5738-dfa4-11e5-831a-2870691a73c5.png)
After:
![screenshot from 2016-03-01 11-42-05](https://cloud.githubusercontent.com/assets/10885517/13436333/548d6296-dfa4-11e5-92fb-2b8ebe12c581.png)

nhomar commented on website_variants_extra/static/src/less/website.less:
```diff
@@ -0,0 +1,71 @@
+        width: 30px;
+    }
+    .website-sale-actions .icon-quote {
+        width: 20px;
```
Usar less:

``` less
.website-sale-actions {
  .custom-icon {
    width: 30px;
  }  
  .icon-quote {
    width: 30px;
  }
}
```

nhomar commented on website_variants_extra/static/src/js/test_product_name_with_variants.js:
```diff
@@ -0,0 +1,88 @@
+    });
+
+}());
+
```
No logro ver donde pruebas el "Quantity of comments."

nhomar commented on website_variants_extra/static/src/less/website.less:
```diff
@@ -0,0 +1,71 @@
+    .product_header{
+        color: @link-color;
+    }
+    hr{
```
`hr {` Sin el espacio esto dará problemas tarde o temprano.

nhomar commented on website_variants_extra/static/src/less/website.less:
```diff
@@ -0,0 +1,71 @@
+    .js_quantity.form-control {
+        background-color: white;
+    }
+    #add_to_cart{
```
intenta usar clases no #

nhomar commented on website_variants_extra/views/templates.xml:
```diff
@@ -3,135 +3,135 @@
-		</div>
+                </div>
+                </div>
+		  </xpath>
```
identaciones limpias pliz.

nhomar commented:
@gmlp mucho mejor... solo comentarios menores.
